plugins/pay.c

```diff
@@ -1043,6 +1043,10 @@ static struct command_result *json_pay(struct command *cmd,
 				    "Invalid bolt11: %s", fail);
 	}
 
+	if (b11->chain != chainparams) {
+		return command_fail(cmd, PAY_ROUTE_NOT_FOUND, "Invoice is for another network %s", b11->chain->network_name);
```

Comment:
Notice that `b11->chain` may be NULL if we don't recognize the network name.

https://github.com/ElementsProject/lightning/blob/0985c6e219f6d3300853b640b5b29c15fb21358f/bitcoin/chainparams.c#L233-L241

As is a user could be tricked into parsing an invoice for an unknown network and then crash. Please use the bip173 prefix extracted from the invoice instead.

Reply:
Oh, right, that's true.
But that would require parsing the bolt11 twice and if the chain is unknown, there's really nothing else to do.
I'm going to simply check if it's null a return an unknown network error if so, that seems simpler.